Modules/Core/Common/include/itkBoundingBox.h

```diff
@@ -90,7 +90,10 @@ class ITK_TEMPLATE_EXPORT BoundingBox : public Object
   /** Hold on to the type information specified by the template parameters. */
   using PointIdentifier = TPointIdentifier;
   using CoordinateType = TCoordinate;
-  using CoordRepType = CoordinateType;
+#ifndef ITK_FUTURE_REMOVE_LEGACY
+  using CoordRepType [[deprecated("ITK 6 discourages using `CoordRepType`. Please use `CoordinateType` instead!")]] =
+    CoordinateType;
+#endif
```

Comment:
I'm not sure but I'm afraid that some users might get annoyed by this deprecated-warning. Especially those users who have intentionally enabled legacy support. What do you think?

Reply:
I'm thinking, maybe we should have a FUTURE_DEPRECATED macro, roughly as follows:

```cpp
#ifdef ITK_LEGACY_REMOVE
#define FUTURE_DEPRECATED [[deprecated]]
#else
#define FUTURE_DEPRECATED
#endif
```
To be used as follows:

```cpp
    using CoordRepType FUTURE_DEPRECATED = CoordinateType;
```

So that the warning only appears when users enable LEGACY_REMOVE. What do you think?

Reply:
There are compiler flags to control warnings.

I now notice: is the correct ifdef `ITK_FUTURE_LEGACY_REMOVE`? You used `ITK_FUTURE_REMOVE_LEGACY`.

Reply:
That `FUTURE_DEPRECATED` macro sounds good to me.

Reply:
On the other hand, I just hope that this PR can be processed quickly, before Hans (@hjmjohnson) submits another mega-PR  😸 His PR #4996 already caused some conflicts locally, when I was preparing this PR.